### PR TITLE
Add a moderation cog

### PR DESCRIPTION
**Pull request by TildeBeta**
Current features:
 - Kicking
 - Banning
 - Softbanning
 - Unbanning

Changes #4

### CODE REVIEW

File: moderation.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import asyncio
+import re
+
+import discord
+from discord.ext import commands
+
+
+class HierarchyMember(commands.MemberConverter):
+    """
+    A member converter that respects Discords role hierarchy system.
```
Comment (Fuyukai) [inline moderation.py:10]: Discord's

File: moderation.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+class BannedUser(commands.Converter):
+    """
+    A converter for members who have been banned.
+    """
+    async def convert(self, ctx, arg):
+        bans = [entry.user for entry in await ctx.guild.bans()]
+
+        try:
+            user = discord.utils.get(bans, id=int(arg))
```
Comment (Fuyukai) [inline moderation.py:45]: Why is this flattened and then filtered? Why not loop over and break if you hit the user?

**Review by Fuyukai — APPROVED** [review]

**Comment by indiv0** [comment]
@SunDwarf ??? the typo didn't get fixed.

**Comment by Fuyukai** [comment]
I didn't care enough to block the PR for just the typo

**Pull request merged**